lib/bookstore-scrapper/helpers/getTextbookInformation.ts

```diff
@@ -0,0 +1,180 @@
+const endCourseNamePath =
+  "&filter%5Bexact_match%5D=1&filter%5Bitem_type%5D=3&filter%5Bonly_active%5D=1&filter%5B3%5D=1#acalog_template_course_filter";
+const endCourseParseOne = "  opens a new window";
+const endCourseParseTwo = "</a><span";
+const finalPageText = "Other Courses";
+
+// Values to parse price from book page
+const startParsePrice = 'value="';
+const midParsePrice = '" onclick="toggle';
+const endParsePrice = "new";
+const quoteParsePrice = "'";
+const notAvailable = null;
+
+export interface CampusStoreEntry {
+  book: Book;
+  course: Course;
+  dept: Dept;
+}
+
+export const getTextbookInformation = async (
+  link: string,
+  program: string,
+  term: string,
+  dept: string,
+  course: string
+): Promise<CampusStoreEntry[]> => {
+  // Array to return
+  const campusStoreEntries: CampusStoreEntry[] = [];
+
+  // Get HTML source code of page
+  let bookHTML: string = (await axiosRequest(link)) as string;
+
+  // Check if page has additonal recommended supplies
+  const suppliesIndex: number = bookHTML.indexOf(parseEndBookHTML);
+  if (suppliesIndex > 0) bookHTML = bookHTML.substring(0, suppliesIndex);
+
+  // Split the different books on the page into parts of an array
+  let bookSplit = bookHTML.split(parseISBN);
+
+  // Find out the department
+  const deptSplit = dept.split("-");
+  let deptName = "";
+  const deptAbv = deptSplit[0].trim();
+  if (deptSplit.length === 1) {
+    deptName = deptAbv;
+  } else {
+    deptName = deptSplit[1].trim();
+  }
+  const deptData: Dept = {
+    id: randomUUID(),
+    name: deptName,
+    abbreviation: deptAbv,
+  };
+
+  // Find out the course
+  const courseName = await getCourseName(deptAbv, course);
+  const courseData: Course = {
+    id: randomUUID(),
+    name: courseName,
+    code: course,
+    term: term,
+    deptId: deptData.id,
+  };
+
+  // Create the book data object for each book
+  for (let book = 1; book < bookSplit.length; book++) {
+    const isbn = bookSplit[book].substring(0, 13);
+
+    const campusStorePrice = getBookPrice(bookHTML, isbn);
+
+    if (JSON.stringify(campusStoreEntries).includes(isbn)) {
+      continue;
+    }
+
+    let name = bookSplit[book].substring(
+      bookSplit[book].indexOf(parseBookName) + 5,
+      bookSplit[book].length
+    );
+    name = name.substring(0, name.indexOf(endParseSlash));
+
+    const imageUrl = `${bookImagePath}${isbn}`;
+
+    const googleBooksId = await getGoogleBooksId(isbn);
+
+    const bookData: Book = {
+      id: randomUUID(),
+      isbn,
+      name,
+      imageUrl,
+      googleBooksId,
+      isCampusStoreBook: true,
+      campusStorePrice,
+    };
+
+    const campusStoreEntry: CampusStoreEntry = {
+      book: bookData,
+      course: courseData,
+      dept: deptData,
+    };
+
+    campusStoreEntries.push(campusStoreEntry);
+  }
+  return campusStoreEntries;
+};
+
+const getCourseNameHTML = async (): Promise<string> => {
+  let page: number = 1;
```

Comment:
No need to explicitly add a type annotation here, the type can be inferred by TypeScript and humans.

Reply:
Sounds good. Will change that.